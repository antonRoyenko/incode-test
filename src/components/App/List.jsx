import React from 'react';
import { Button, Card, Image } from 'semantic-ui-react';

export const List = (props) => {
    if (props.users[0].user !== 'no data') {
        return (
            <ul>
                {props.users.map((item, key) => (
                    <li key={key} onClick={() => props.setActive(item.user.id)}>
                        <Card>
                            <Card.Content>
                                <Image floated='left' size='mini' src={item.user.general.avatar} />
                                <Card.Header>{item.user.general.firstName} {item.user.general.lastName}</Card.Header>
                                <Card.Meta>{item.user.job.title} - {item.user.job.company}</Card.Meta>
                            </Card.Content>
                            <Card.Content extra>
                                <div className='ui two buttons'>
                                    <Button basic color='grey'>
                                        Detail
                                    </Button>
                                </div>
                            </Card.Content>
                        </Card>
                        </li>
                    )
                )}
            </ul>
        )
    } else {
        return (
            <ul>
                <li>
                    <Card>
                        <Card.Content>
                            <Image floated='left' size='mini' src='https://react.semantic-ui.com/images/avatar/large/steve.jpg' />
                            <Card.Header>Sorry</Card.Header>
                            <Card.Meta>On request {props.request} ... nothing was found. </Card.Meta>
                        </Card.Content>
                    </Card>
                </li>
            </ul>
        )
    }

};
