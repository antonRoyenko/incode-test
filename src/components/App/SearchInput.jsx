import React from 'react';

export const SearchInput = (props) => (
    <div>
        <form>
            <label>
                Name:
                <input type="text" value={props.value} onChange={props.handleChange} />
            </label>
            <input type="submit" value="Submit" />
        </form>
    </div>
);
