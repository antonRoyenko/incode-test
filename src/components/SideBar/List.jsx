import React from 'react';

export const List = (props) => {
    console.log(props);
    return (<div>
        test
    </div>)
};
