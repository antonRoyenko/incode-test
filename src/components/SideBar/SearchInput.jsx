import React from 'react';

export const SearchInput = () => (
    <div>
        <input type="text"/>
    </div>
);
