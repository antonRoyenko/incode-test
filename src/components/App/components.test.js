import React from 'react';
import {shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { SearchInput } from './SearchInput';
import { List } from './List';

const data = [{
    user: {
        "id": 20,
        "general": {
            "firstName": "Elmer",
            "lastName": "D'Amore",
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/mfacchinello/128.jpg"
        },
        "job": {
            "company": "Schaden Group",
            "title": "Regional Brand Strategist"
        },
        "contact": {
            "email": "Margarett57@gmail.com",
            "phone": "1-236-341-6098 x2838"
        },
        "address": {
            "street": "26952 Welch Course",
            "city": "Lake Carmella land",
            "zipCode": "85577-5136",
            "country": "Israel"
        }
    }
}];

configure({ adapter: new Adapter() });

describe('SearchInput', () => {
    const props = { value: 'test' };
    const searchInput = shallow(<SearchInput {...props}/>);

    it('renders search input', () => {
        expect(searchInput).toMatchSnapshot();
    });

    it('create input', () => {
        expect(searchInput.find('.search-input').exists()).toBe(true);
    });

    describe('when user types in input', () => {
        const userValue = 'test';

        beforeEach(() => {
            searchInput.find('.search-input').simulate('change', { target: { value: userValue } })
        });

        it('updatet input state', () => {
            expect(searchInput.find('.search-input').props().value).toEqual(userValue);
        });
    })
});

describe('List', () => {
    const props = { value: 'test' };
    const list = shallow(<List users={data} />);

    it('renders users list', () => {
        expect(list.props()).toMatchSnapshot();
    });

    it('exist list', () => {
        expect(list.find('li').exists()).toBe(true);
    });
});
