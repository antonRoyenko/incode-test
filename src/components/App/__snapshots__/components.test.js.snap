// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders users list 1`] = `
Object {
  "children": Array [
    <li
      onClick={[Function]}
    >
      <Card>
        <CardContent>
          <Image
            as="img"
            floated="left"
            size="mini"
            src="https://s3.amazonaws.com/uifaces/faces/twitter/mfacchinello/128.jpg"
            ui={true}
          />
          <CardHeader>
            Elmer
             
            D'Amore
          </CardHeader>
          <CardMeta>
            Regional Brand Strategist
             - 
            Schaden Group
          </CardMeta>
        </CardContent>
        <CardContent
          extra={true}
        >
          <div
            className="ui two buttons"
          >
            <Button
              as="button"
              basic={true}
              color="grey"
              role="button"
            >
              Detail
            </Button>
          </div>
        </CardContent>
      </Card>
    </li>,
  ],
}
`;

exports[`SearchInput renders search input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput
    value="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Input
        className="search-input"
        icon={
          Object {
            "circular": true,
            "link": true,
            "name": "search",
          }
        }
        placeholder="Search..."
        type="text"
        value="test"
      />,
      "className": "search-block",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "search-input",
        "icon": Object {
          "circular": true,
          "link": true,
          "name": "search",
        },
        "onChange": undefined,
        "placeholder": "Search...",
        "type": "text",
        "value": "test",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Input
          className="search-input"
          icon={
            Object {
              "circular": true,
              "link": true,
              "name": "search",
            }
          }
          placeholder="Search..."
          type="text"
          value="test"
        />,
        "className": "search-block",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "search-input",
          "icon": Object {
            "circular": true,
            "link": true,
            "name": "search",
          },
          "onChange": undefined,
          "placeholder": "Search...",
          "type": "text",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
